### ISSUE TITLE
Use Correct Foreground Colors for Text and Icons in Dark and Light Modes

### ISSUE THREAD

**Issue by alperozturk96** [nextcloud/server#46698]
### Problem

@nextcloud/designers @nextcloud/designers-community 

When dark mode enabled, icons and texts colors are not providing good contrast against background color. Recently modified files are highlighted well in dark mode, but files modified some time ago use a grey color that doesn't provide good contrast.

![Screenshot 2024-07-23 at 09 44 22](https://github.com/user-attachments/assets/af888aa2-39d7-40e9-b698-216668cf885c)
![Screenshot 2024-07-23 at 09 44 40](https://github.com/user-attachments/assets/86928919-ea81-4076-bc4b-6abfeb345a62)

### Solution

Corresponding foreground color must be used for icons and texts.

**Comment by tobiasKaminsky** [comment]
Android: https://github.com/nextcloud/android/issues/13309
Android notes: https://github.com/nextcloud/notes-android/issues/2300

**Comment by susnux** [comment]
This is done on purpose:
Older files get slightly less intense colors to reflect their age.
This color is based on the modification time and range from maxcontrast to main text. 

CC @nextcloud/designers if we should remove this feature.

**Comment by nimishavijay** [comment]
> This is done on purpose: Older files get slightly less intense colors to reflect their age. This color is based on the modification time and range from maxcontrast to main text.

I would vote for removing this in the interest of maintaining consistency of colors. 

We can start off with removing the range of colors and sticking to `--color-main-text` for recent files and `--color-text-maxcontrast` for older files as the result of the color mix is often too subtle to notice anyway.

**Comment by skjnldsv** [comment]
> I would vote for removing this in the interest of maintaining consistency of colors.

We do have the same for file sizes. The larger the file, the stronger the size color will be. This is an old feature that many uses. :warning:

**Comment by marcoambrosini** [comment]
I think that everything should be of the same color, same as the file name. I don't think the feature is helpful because it's not easy to understand what these color difference indicate, as @alperozturk96 misinterpretation of it demonstrates.

**Comment by jancborchardt** [comment]
> We do have the same for file sizes. The larger the file, the stronger the size color will be. This is an old feature that many uses. ⚠️

Exactly, as @susnux and @skjnldsv said this was intentionally added a long time ago and is a nice differentiator for us compared to other solutions as well.
- It helps distinguish newer (aka more relevant) files even if the list is sorted alphabetically
- It also directly highlights larger (using more space if you need to clean up for example)

So no, we should not remove this. (FYI @nimishavijay @marcoambrosini) And as said the contrast is fine since the lowest we go is `color-text-maxcontrast`.

We also do a similar thing with the file extensions in the file list, because they are not as important as the file name itself. It helps to focus.
